boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -100,6 +100,24 @@ CXXFLAGS := $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) \
 CPPFLAGS := $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS)
 AFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
+ifeq ($(CONFIG_LIBCXX),y)
+  CXXFLAGS += -std=c++14 -D_LIBCPP_STD_VER=14
+
+  ifeq ($(CONFIG_HOST_MACOS),y)
+    # macOS uses libc++abi
+    CXXFLAGS += -DLIBCXX_BUILDING_LIBCXXABI
+  else
+    # Linux C++ ABI seems vary.
+    # Probably __GLIBCXX__ is the best bet.
+    # XXX what to do for windows?
+    CXXFLAGS += -D__GLIBCXX__
+  endif
+
+  ifneq ($(CONFIG_CXX_EXCEPTION),y)
+    CXXFLAGS += -D_LIBCPP_NO_EXCEPTIONS
```

Comment:
don't need libc++ will detect the setting automatically:
```
#if !__has_feature(cxx_exceptions)
#  define _LIBCPP_NO_EXCEPTIONS
#endif
```

Reply:
i forgot why i added this. probably i will remove this.